spec/index.bs

```diff
@@ -1019,11 +1019,37 @@ conditions.
 <script type=idl>
 typedef (Uint32Array or DOMString) GPUShaderCode;
 
+typedef any GPUShaderSource;
+callback GPUShaderTransformFunction = GPUShaderCode (GPUShaderSource source);
+
 dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
-    required GPUShaderCode code;
+    required GPUShaderSource source;
+    GPUShaderTransformFunction transform;
 };
 </script>
 
+{{GPUShaderModuleDescriptor}} has the following internal slots:
```

Comment:
The descriptor is just a dictionary so it doesn't make sense to me to put the internal slot here; I think we can just make it the 'return value' of the "resolving GPUShaderCode" algorithm. (we'll have to do something with that return value later on)

Reply:
Would it make sense to have the algorithm describe GPUDevice.createShaderModule and have [[code]] be an internal slot of GPUShaderModule?

Reply:
I think if you did this inside createShaderModule, it would only need to exist as a local variable inside that algorithm, and not stored off anywhere.

Reply:
Updated.